packages/cli/src/cmds/env.ts

```diff
@@ -60,6 +66,87 @@ export default class Env extends YargsCommand {
   }
 }
 
+class EnvAdd extends YargsCommand {
+  public readonly commandHead = `add`;
+  public readonly command = `${this.commandHead} <name>`;
+  public readonly description =
+    "Add a new environment by copying from current active environment or the specified environment.";
+  public params: { [_: string]: Options } = {};
+
+  public builder(yargs: Argv): Argv<any> {
+    // TODO: support --details
+    this.params = HelpParamGenerator.getYargsParamForHelp(Stage.createEnv);
+    yargs.positional("name", {
+      description: "The new environment name",
+      type: "string",
+      require: true,
+    });
+    return yargs.version(false).options(this.params);
+  }
+
+  public async runCommand(args: { [argName: string]: string }): Promise<Result<null, FxError>> {
+    const projectDir = args.folder || process.cwd();
+    // args.name always exists because or `require: true` in builder
+    const targetEnv = args.name as string;
+    const argsEnv = args.env as string | undefined;
```

Comment:
nit: would be more straightforward:
```suggestion
    const argsEnv = args.env as string;
```

Reply:
The `--env` in `teamsfx add` command is optional, so it is not always a string.